### PR TITLE
adding HTML body to emails

### PR DESCRIPTION
**Pull request by kpsf**

### CODE REVIEW

**Comment by kpsf** [comment]
Ready for testing

File: src/main/java/com/desk/java/apiclient/model/Interaction.java [hunk 1]
```diff
@@ -51,6 +51,8 @@
     private String client_type;
     private String subject;
     private boolean hidden;
+    private String body_html;
+    private String body_text;
```
Comment (tstachl) [inline src/main/java/com/desk/java/apiclient/model/Interaction.java]: not needed

File: src/main/java/com/desk/java/apiclient/model/Interaction.java [hunk 2]
```diff
@@ -162,15 +164,23 @@ public String getBody()
         return body;
     }
 
-    /**
-     * Set the Message body.
-     * @param string Message body.
-     */
+
```
Comment (tstachl) [inline src/main/java/com/desk/java/apiclient/model/Interaction.java]: not needed

File: src/main/java/com/salesforce/scmt/utils/DeskJsonMapUtil.java [hunk 3]
```diff
@@ -687,7 +687,24 @@ public static void deskInteractionToSalesforceJsonMaps(DeskUtil deskUtil,
             deployResponse.addErrorWithId(msg, interaction.getId());
             interaction.setBody(interaction.getBody().substring(0, SalesforceConstants.EMAIL_BODY_LENGTH));
         }
-        mapObj.put(EmailMessageFields.HtmlBody, interaction.getBody());
+
```
Comment (tstachl) [inline src/main/java/com/salesforce/scmt/utils/DeskJsonMapUtil.java]: one line break is enough

File: src/main/java/com/salesforce/scmt/utils/DeskJsonMapUtil.java [hunk 4]
```diff
@@ -687,7 +687,24 @@ public static void deskInteractionToSalesforceJsonMaps(DeskUtil deskUtil,
...
+
+        if (interaction.getHtmlBody() != null && interaction.getHtmlBody().length() > SalesforceConstants.EMAIL_BODY_LENGTH)
```
Comment (tstachl) [inline src/main/java/com/salesforce/scmt/utils/DeskJsonMapUtil.java]: can we create a method for this using it with both the body and the html body?

File: src/main/java/com/salesforce/scmt/utils/DeskJsonMapUtil.java [hunk 5]
```diff
@@ -687,7 +687,24 @@ public static void deskInteractionToSalesforceJsonMaps(DeskUtil deskUtil,
...
+        {
+            String msg = String.format(
+                    "Interaction body exceeds max length and will be truncted! ");
+            Utils.log(msg);
+            deployResponse.addErrorWithId(msg, interaction.getId());
+            interaction.setHtmlBody(interaction.getHtmlBody().substring(0, SalesforceConstants.EMAIL_BODY_LENGTH));
+        }
+
+
+        if (interaction.getHtmlBody() !=null && interaction.getHtmlBody() != "") {
```
Comment (tstachl) [inline src/main/java/com/salesforce/scmt/utils/DeskJsonMapUtil.java]: We should do something like this: `str != null && !str.isEmpty()` as `" " != ""`

File: src/main/java/com/salesforce/scmt/utils/DeskUtil.java [hunk 6]
```diff
@@ -1956,8 +1956,8 @@ private static SObject deskInteractionToSalesforceEmail(Interaction interaction,
         so.addField(EmailMessageFields.HasAttachment, (interaction.getLinks().getAttachments().getCount() > 0));
         
         // The body of the email in HTML format.
-        so.addField(EmailMessageFields.HtmlBody, interaction.getBody());
-        
+        so.addField(EmailMessageFields.HtmlBody, interaction.getHtmlBody());
```
Comment (tstachl) [inline src/main/java/com/salesforce/scmt/utils/DeskUtil.java]: Is this change still needed?

File: src/main/java/com/desk/java/apiclient/model/Interaction.java [hunk 7]
```diff
@@ -162,15 +164,23 @@ public String getBody()
...
+    {
+        return body_html;
+    }
+
+    public void setHtmlBody(String string)
+    {
+        this.body_html = string;
+    }
+
+
```
Comment (tstachl) [inline src/main/java/com/desk/java/apiclient/model/Interaction.java]: one line brake is enough
